openfe/protocols/openmm_rbfe/equil_rbfe_methods.py

```diff
@@ -803,15 +356,29 @@ def run(self, dry=False, verbose=True, basepath=None) -> dict[str, Any]:
 
         # 0. General setup and settings dependency resolution step
 
-        # a. check equilibration and production are divisible by n_steps
-        settings = self._inputs['settings']
+        # a. check timestep correctness + that 
+        # equilibration & production are divisible by n_steps
+        prototol_settings: RelativeLigandProtocolSettings = self._inputs['settings']
         stateA = self._inputs['stateA']
         stateB = self._inputs['stateB']
-        mapping = self._inputs['mapping']
-
-        sim_settings = settings.simulation_settings
-        timestep = settings.integrator_settings.timestep
-        mc_steps = settings.integrator_settings.n_steps.m
+        mapping = self._inputs['ligandmapping']
+
+        forcefield_settings: settings.OpenMMSystemGeneratorFFSettings = prototol_settings.forcefield_settings
+        thermo_settings: settings.ThermoSettings = prototol_settings.thermo_settings
+        alchem_settings: AlchemicalSettings = prototol_settings.alchemical_settings
+        system_settings: SystemSettings = prototol_settings.system_settings
+        solvation_settings: SolvationSettings = prototol_settings.solvation_settings
+        sampler_settings: AlchemicalSamplerSettings = prototol_settings.alchemical_sampler_settings
+        sim_settings: SimulationSettings = prototol_settings.simulation_settings
+        timestep = prototol_settings.integrator_settings.timestep
+        mc_steps = prototol_settings.integrator_settings.n_steps.m
+
+        # is the timestep good for the mass?
```

Comment:
Is there a way to do this at the validator level? It feels like there should be, but I couldn't find a way to use a validator to validate across two pydantic classes as attributes of a model.

Reply:
The answer _should_ be to use a root_validator in the pydantic settings child class.

_However_, applying validators on nested models is all kinds of broken in pydantic right now (it's meant to be fixed in the v2.0 release). We might just have to keep there here until v2.0 gets released properly.

Reply:
yeah I think we won't be able to get validators to do all the heavy lifting, but they'll at least do a good level of sanity checking